django/core/checks/urls.py

```diff
@@ -19,22 +19,13 @@ def check_resolver(resolver):
     """
     Recursively check the resolver.
     """
-    from django.urls import RegexURLPattern, RegexURLResolver
     warnings = []
     for pattern in resolver.url_patterns:
-        if isinstance(pattern, RegexURLResolver):
-            warnings.extend(check_include_trailing_dollar(pattern))
-            # Check resolver recursively
-            warnings.extend(check_resolver(pattern))
-        elif isinstance(pattern, RegexURLPattern):
-            warnings.extend(check_pattern_name(pattern))
-        else:
-            # This is not a url() instance
+        check_method = getattr(pattern, "check", None)
+        if check_method is None:
             warnings.extend(get_warning_for_invalid_pattern(pattern))
```

Comment:
Hadn't thought about that. Should probably read a bit more about URL resolvers and the like.
Should I be using `getattr` or is exception handling recommended given the EAFP style? Is django more on the LBYL side?